drivers/sensor/isl29125/isl29125_trigger.c

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2020 Intive
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/i2c.h>
+#include <sys/byteorder.h>
+#include <sys/util.h>
+#include <kernel.h>
+#include <logging/log.h>
+
+#include "isl29125.h"
+
+LOG_MODULE_DECLARE(ISL29125, CONFIG_SENSOR_LOG_LEVEL);
+
+static int isl29125_write16(struct device *i2c_dev, u8_t reg, u16_t data)
+{
+	u8_t buf[3];
+
+	data = sys_cpu_to_le16(data);
+	buf[0] = reg;
+	buf[1] = data & 0xFF;
+	buf[2] = data >> 8;
+	int result = i2c_write(i2c_dev, buf, 3, ISL29125_I2C_ADDR);
+
+	if (result) {
+		LOG_ERR("Error writing register %d at Addr:0x%x", reg,
+			ISL29125_I2C_ADDR);
+	}
+
+	return result;
+}
+
+int isl29125_attr_set(struct device *dev, enum sensor_channel chan,
+		      enum sensor_attribute attr,
+		      const struct sensor_value *val)
+{
+	struct isl29125_data *drv_data = dev->driver_data;
+
+	drv_data->dev_config_3 &= ~(ISL29125_CFG3_TH_IRQ_MASK);
+
+	switch (chan) {
+	case SENSOR_CHAN_RED:
+		drv_data->dev_config_3 |= ISL29125_CFG3_R_INT;
+		LOG_INF("Threshold for channel RED");
+		break;
+	case SENSOR_CHAN_GREEN:
+		drv_data->dev_config_3 |= ISL29125_CFG3_G_INT;
+		LOG_INF("Threshold for channel GREEN");
+		break;
+	case SENSOR_CHAN_BLUE:
+		drv_data->dev_config_3 |= ISL29125_CFG3_B_INT;
+		LOG_INF("Threshold for channel BLUE");
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	drv_data->dev_config_3 &= ~(ISL29125_CFG3_INT_MASK);
+	drv_data->dev_config_3 |= ISL29125_CFG3_INT_PRST8;
+
+	if (attr == SENSOR_ATTR_UPPER_THRESH) {
+		LOG_INF("Upper threshold set to: %d", val->val1);
+		if (isl29125_write16(drv_data->i2c, ISL29125_THRESHOLD_HL,
+				     val->val1) != 0) {
+			LOG_ERR("Threshold set failed");
+			return -EIO;
+		}
+
+	} else if (attr == SENSOR_ATTR_LOWER_THRESH) {
+		LOG_INF("Lower threshold set to: %d", val->val1);
+		if (isl29125_write16(drv_data->i2c, ISL29125_THRESHOLD_LL,
+				     val->val1) != 0) {
+			LOG_ERR("Threshold set failed");
+			return -EIO;
+		}
+	} else {
+		return -EINVAL;
+	}
+
+	return isl29125_set_config(drv_data);
+}
+
+static void isl29125_gpio_callback(struct device *dev, struct gpio_callback *cb,
+				   u32_t pins)
+{
+	ARG_UNUSED(pins);
+
+	struct isl29125_data *drv_data =
+		CONTAINER_OF(cb, struct isl29125_data, gpio_callback);
+	gpio_pin_disable_callback(dev, DT_INST_0_ISIL_ISL29125_INT_GPIOS_PIN);
```

Comment:
Heads up that the gpio API is about to change, and you will need to update this driver accordingly. Please see #21789 for details